bigquery/google/cloud/bigquery/_helpers.py

```diff
@@ -28,10 +28,22 @@
 
 _RFC3339_MICROS_NO_ZULU = '%Y-%m-%dT%H:%M:%S.%f'
 
-# Canonical formats for timestamps in BigQuery
+# Canonical formats for timestamps in BigQuery.
+# Timestamps in an API response will always be in UTC, but they
+# can vary in other ways. The separator between the date and time
+# can be 'T' or ' ', the microseconds may be present or not, and
+# the UTC timezone may be formatted as Z or +00:00. See:
 # g.co/cloud/bigquery/docs/reference/standard-sql/data-types#timestamp-type
-_BIGQUERY_TIMESTAMP = '%Y-%m-%d %H:%M:%S%z'
-_BIGQUERY_TIMESTAMP_MICROS = '%Y-%m-%d %H:%M:%S.%f%z'
+_BIGQUERY_TIMESTAMP_FORMATS = [
+    '%Y-%m-%d %H:%M:%S+00:00',
+    '%Y-%m-%d %H:%M:%SZ',
+    '%Y-%m-%d %H:%M:%S.%f+00:00',
+    '%Y-%m-%d %H:%M:%S.%fZ',
+    '%Y-%m-%dT%H:%M:%S+00:00',
+    '%Y-%m-%dT%H:%M:%SZ',
+    '%Y-%m-%dT%H:%M:%S.%f+00:00',
+    '%Y-%m-%dT%H:%M:%S.%fZ',
+]
```

Comment:
- If we *know for certain* that the backend will send only UTC timestamps, we can just crack off the trailing `Z` / `+00:00`.

- Given that the spec is about the formats the back-end will *accept*, they should be able to promise that they will send *back* the timestamp using exactly *one* format (` ` vs. `T` separator, `Z` vs. `+00:00`).  [Postel's Law](https://en.wikipedia.org/wiki/Robustness_principle) applies.

Reply:
(1) Okay. I'll do some chopping.

(2) Agreed, but I wasn't able to get the backend team to commit. The protos promise one format, but the actual system behavior is different.

Reply:
Chopping done. I did some string replacing too for `' '` vs `'T'`. Wasn't as complicated as I expected.